crates/legacy/src/service.rs

```diff
@@ -388,6 +395,37 @@ impl<Types: NodeType> GlobalState<Types> {
         .await
     }
 
+    // get transaction status
+    // return one of "pending", "sequenced", "rejected" or "unknown"
+    pub fn claim_tx_status(
+        &self,
+        txn_hash: Commitment<<Types as NodeType>::Transaction>,
+    ) -> Result<TransactionStatus, BuildError> {
+        if let Some(status) = self.tx_status.get(&txn_hash) {
+            Ok(status.clone())
+        } else {
+            Ok(TransactionStatus::Unknown)
+        }
+    }
+
+    pub fn set_tx_status(
+        &mut self,
+        txn_hash: Commitment<<Types as NodeType>::Transaction>,
+        txn_status: TransactionStatus,
+    ) -> Result<(), BuildError> {
+        if self.tx_status.contains_key(&txn_hash) {
```

Comment:
I don't think we want to allow switching txn status from `Sequenced` to `Pending`/`Rejected`, or duplicate submissions of already-sequenced transactions will put them in these states forever